presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/file/TableMetadata.java

```diff
@@ -75,7 +78,7 @@ public TableMetadata(
         this.dataColumns = ImmutableList.copyOf(requireNonNull(dataColumns, "dataColumns is null"));
         this.partitionColumns = ImmutableList.copyOf(requireNonNull(partitionColumns, "partitionColumns is null"));
         this.parameters = ImmutableMap.copyOf(requireNonNull(parameters, "parameters is null"));
-
+        this.preferredOrderingColumns = ImmutableList.copyOf(requireNonNull(preferredOrderingColumns.orElse(ImmutableList.of()), "preferredOrderingColumns is null"));
```

Comment:
Shall we make the stack variable the same as the constructor param? Having one with Optional and the other not is kinda confusing. If empty list means no preferred sorting, then we can just drop Optional.

Reply:
That was exactly what I tried and then realized since it is also a `JsonCreator`, you will hit NPE when you decode something without this field :(